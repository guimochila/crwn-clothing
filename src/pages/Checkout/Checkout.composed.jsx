import React from 'react';
import { Query } from 'react-apollo';

import Checkout from './Checkout';
import { GET_CART_ITEMS_AND_TOTAL } from '../../graphql/queries';

const CheckoutComposed = () => (
  <Query query={GET_CART_ITEMS_AND_TOTAL}>
    {({ data: { cartItems, cartTotal } }) => (
      <Checkout cartItems={cartItems} total={cartTotal} />
    )}
  </Query>
);

export default CheckoutComposed;
