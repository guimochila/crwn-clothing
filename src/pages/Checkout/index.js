import CheckoutComposed from './Checkout.composed';

export default CheckoutComposed;
