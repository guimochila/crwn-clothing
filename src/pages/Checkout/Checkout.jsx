import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { useInView } from 'react-intersection-observer';

import './Checkout.scss';
import {
  selectCartItems,
  selectCartTotal,
} from '../../store/cart/cart.selectors';
import CheckoutItem from '../../components/CheckoutItem';
import StripeButton from '../../components/StripeButton';

export function Checkout({ cartItems, total }) {
  const [ref, inView] = useInView();

  return (
    <div className="checkout-page">
      <div className="checkout-header">
        <div className="header-block">
          <span>Product</span>
        </div>
        <div className="header-block">
          <span>Description</span>
        </div>
        <div className="header-block">
          <span>Quantity</span>
        </div>
        <div className="header-block">
          <span>Price</span>
        </div>
        <div className="header-block">
          <span>Remove</span>
        </div>
      </div>
      {cartItems.map(cartItem => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      ))}
      <div ref={ref} />
      <div className={`total ${inView ? '' : 'total-not-visible'}`}>
        <span>TOTAL: ${total}</span>
        <StripeButton price={total} />
      </div>
      <div className="test-warning">
        *Please use the following test credit card for payment <br />
        4242-4242-4242-4242 - Exp: 01/20 CVV: 123
      </div>
    </div>
  );
}

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

export default connect(mapStateToProps)(Checkout);
