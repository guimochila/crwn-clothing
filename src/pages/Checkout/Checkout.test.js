import React from 'react';
import { shallow } from 'enzyme';

import { Checkout } from './Checkout';

describe('Checkout Page', () => {
  test('should render Checkout page', () => {
    let mockProps = {
      cartItems: [],
      total: 100,
    };
    let wrapper = shallow(<Checkout {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
