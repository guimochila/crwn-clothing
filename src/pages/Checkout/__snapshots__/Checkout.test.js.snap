// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Page should render Checkout page 1`] = `
<div
  className="checkout-page"
>
  <div
    className="checkout-header"
  >
    <div
      className="header-block"
    >
      <span>
        Product
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Description
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Quantity
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Price
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Remove
      </span>
    </div>
  </div>
  <div />
  <div
    className="total total-not-visible"
  >
    <span>
      TOTAL: $
      100
    </span>
    <StripeButton
      price={100}
    />
  </div>
  <div
    className="test-warning"
  >
    *Please use the following test credit card for payment 
    <br />
    4242-4242-4242-4242 - Exp: 01/20 CVV: 123
  </div>
</div>
`;
